docs/sdk/auth/fragments/android/how-it-works.md

```diff
@@ -51,23 +59,27 @@ The `AWSMobileClient` will return valid JWT tokens from your cache immediately i
 
 If the Refresh tokens have expired and you then make call to any AWS service, such as a AppSync GraphQL request or S3 upload, the `AWSMobileClient` will dispatch a state notification that a re-login is required. At this point you can choose to present the user with a login screen, call `AWSMobileClient.default().signIn()`, or perform custom business logic. For example:
 
-```swift
-AWSMobileClient.default().addUserStateListener(self) { (userState, info) in
-            
-            switch (userState) {
-            case .signedOut:
+```java
+AWSMobileClient.getInstance().addUserStateListener(new UserStateListener() {
+    @Override
+    public void onUserStateChanged(UserStateDetails userStateDetails) {
+        switch (userStateDetails.getUserState()){
+            case SIGNED_OUT:
                 // user clicked signout button and signedout
-                print("user signed out")
-            case .signedOutUserPoolsTokenInvalid:
-                print("need to login again.")
-                AWSMobileClient.default().signIn(username: "username", password: "password", completionHandler: { (res, err) in
-                    //...
+                Log.i("userState", "user signed out");
+                break;
+            case SIGNED_OUT_USER_POOLS_TOKENS_INVALID:
+                Log.i("userState", "need to login again.");
+                AWSMobileClient.getInstance().signIn(username, password, null, new Callback<SignInResult>() {
+                    //... 
                 });
```

Comment:
Instead of nesting this, does it make sense to split out a helper `signIn(...)` method? I'm worried that a novice programmer won't know to do this for themselves, if they're just starting with this example -- even after they add/refactor their code.

Reply:
I'm going to circle back on this one